### ISSUE TITLE
Watermarks are not entirely Random

### ISSUE THREAD

**Issue by m472** [microsoft/InvisMark#8]
I was very confused to find a much lower bit accuracy than stated in the paper when evaluating the published model checkpoint on DIV2K using JPEG quality 50.
Instead of 0.995 as stated in Table 3 it was only 0.93 in my tests with 100 bits of random data, which is a very large discrepancy.

I then had a look at the training script and found out the reason for this is the training uses the first 100 Bits of a UUID4. The issue with using UUIDs instead of random binary data is the version identifier of the UUID, which is 6 bits in total and is completely statical. This means the decoder is trained to always output the same bit pattern for this part of the message. Because of this the capacity of the information channel is only 94 bits instead of 100 bits, as the 6 constant bits contain no information (from an information theory standpoint).

The paper states in section 3: "Each image is encoded with randomly generated 100 bit watermark." which is a bit missleading when only 94 bits of the watermark are actually random.
